unlock-app/src/services/walletService.js

```diff
@@ -91,23 +95,58 @@ export default class WalletService extends EventEmitter {
   }
 
   /**
-   * Function which yields the address of the account on the provider or creates a key pair.
+   * Poll to see if account has changed
    */
-  getAccount() {
-    const getOrCreateAccount = this.web3.eth.getAccounts().then(accounts => {
-      if (accounts.length === 0) {
-        return this._createAccount()
-      } else {
-        return Promise.resolve(accounts[0]) // defaults to the first account for now
-      }
-    })
+  async pollForAccountChange(isServer) {
+    if (isServer) return
+    try {
+      await delayPromise(POLLING_INTERVAL)
```

Comment:
Is there any reason why the `delayPromise` would throw? IIRC it does not based on what you sent earlier?